pennylane/devices/default_qubit.py

```diff
@@ -340,6 +343,10 @@ def _apply_operation(self, state, operation):
 
         return self._apply_unitary(state, matrix, wires)
 
+    def _apply_global_phase(self, state, operation: qml.GlobalPhase):
+        """Applies a :class:`~.GlobalPhase` operation to thet state."""
+        return qml.math.exp(1j * operation.data[0]) * state
```

Comment:
Does this still work if the phase is controlled on another qubit?

Reply:
Then operation would not be a `GlobalPhase` operation, it would be a `Controlled` operation.

Reply:
What would the UI be to enact such an operation?

Reply:
`qml.ctrl(qml.GlobalPhase(phi, 0), (1,2,3))`. then the simulator would just use the matrix instead of a more optimized kernel.

Reply:
Ok, that's great then! Thanks